src/go/Rumble/rumble_server.go

```diff
@@ -0,0 +1,394 @@
+    }
+    log.Println("[INFO] Verbosity:", verbose)
+    userId := strings.ReplaceAll(request.UserId, " ", "") // Clear whitespace
+    response, ok := runRumble(port1, port2, jsoniqFileName, request.OutputPath, userId, verbose, &request.SparkConf)
+    log.Printf("[INFO] Response: %#v", response)
+    releasePorts(port1, port2, portsInUseLockFname) // Release reserved ports
+    log.Println("[INFO] Ports released", port1, port2)
+    if !ok {
+        log.Println("[INFO] Fail.")
+        w.WriteHeader(http.StatusBadRequest)
+        _ = json.NewEncoder(w).Encode(response)
+        return
+    }
+    //os.Remove(jsoniqFileName) // Remove temp JSONiq file
+    log.Println("[INFO] Success.")
+    w.WriteHeader(http.StatusOK)
+    _ = json.NewEncoder(w).Encode(response)
+}
+
+// RumbleServerRequestHandler we should only receive POST request
+func RumbleServerRequestHandler(w http.ResponseWriter, r *http.Request) {
+    w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
+    w.Header().Set("Access-Control-Allow-Origin", "*")
+    w.Header().Set("Accept", "application/json")
+    w.Header().Set("Content-Type", "application/json")
+    var request Request
+    decoder := json.NewDecoder(r.Body)
+    err := decoder.Decode(&request)
+    log.Printf("[INFO] Request - Method: %v, Host: %v, RemoteAddr: %v, Args: %v, URI: %v",
+        r.Method, r.Host, r.RemoteAddr, r.URL, r.RequestURI)
+    if err != nil {
+        tempBody, _ := ioutil.ReadAll(r.Body)
+        log.Printf("[ERROR] could not get request body, reason: %v, Body:%#v", err, tempBody)
+        w.WriteHeader(http.StatusBadRequest)
+        _ = json.NewEncoder(w).Encode(Response{
+            ErrorMsg: ErrorMessage{
+                Message:  "Please check your query, json could not be decoded!",
+                SysError: fmt.Sprintf("%v", err),
+            },
+        })
+        return
+    }
+    log.Printf("[INFO] Request Body:%#v", request)
+    // Check user parameters
+    if request.OutputPath == "" || request.UserId == "" || request.Query == "" {
+        w.WriteHeader(http.StatusBadRequest)
+        _ = json.NewEncoder(w).Encode(Response{
+            ErrorMsg: ErrorMessage{
+                Message: "'query', 'output_path' or 'userid' cannot be empty!",
+            },
+        })
+        return
+    }
+    // Set defaults
+    verbose := 0
+    if request.SparkConf.SparkExecutorMemory == "" {
+        request.SparkConf.SparkExecutorMemory = "2g"
+    }
+    if request.SparkConf.SparkExecutorInstances == "" {
+        request.SparkConf.SparkExecutorInstances = "2"
+    }
+    if request.SparkConf.SparkExecutorCores == "" {
+        request.SparkConf.SparkExecutorCores = "2"
+    }
+    if request.SparkConf.SparkDriverMemory == "" {
+        request.SparkConf.SparkDriverMemory = "2g"
+    }
+    processRumbleServerRequest(w, &request, verbose)
+}
+
+func handlers() *mux.Router {
+    router := mux.NewRouter()
+    fs := http.FileServer(http.Dir("./static"))
```

Comment:
replace static dir with new Go feature called embed, see
- https://golang.org/pkg/embed/
- https://blog.carlmjohnson.net/post/2021/how-to-use-go-embed/
- https://golangtutorial.dev/tips/embed-files-in-go/
This is very big deal since it will allow to emded static file content into the executable and we do not need to worry to deploy static dir on a node.